From 7e4f1581b135dd64196c129cdb9603b4c3bdff9d Mon Sep 17 00:00:00 2001
From: Andrew Hsieh <andrewhsieh@google.com>
Date: Mon, 8 Apr 2013 13:38:08 +0800
Subject: [PATCH 4/5] Fix --build-id to not accept next arg

Only the following two forms are allowed
  --build-id             # default to sha1
  --build-id=style       # choose style

Neither swallow next arg, which happen to be "-m" when ld.mcld is invoked
from clang, and cause GNU linker emulation fail

Change-Id: I57f99c0adc112dcc89dd23a778e3932c03f7d413
---
 tools/llvm-mcld/llvm-mcld.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tools/llvm-mcld/llvm-mcld.cpp b/tools/llvm-mcld/llvm-mcld.cpp
index d84fcc6..b5152e6 100644
--- a/tools/llvm-mcld/llvm-mcld.cpp
+++ b/tools/llvm-mcld/llvm-mcld.cpp
@@ -591,8 +591,12 @@ ArgExcludeLIBS("exclude-libs",
                cl::desc("Exclude libraries from automatic export"),
                cl::value_desc("lib1,lib2,..."));
 
+static cl::opt<bool>
+ArgBuildIDDefault("build-id",
+           cl::desc("Request creation of \".note.gnu.build-id\" ELF note section in sha1."));
+
 static cl::opt<std::string>
-ArgBuildID("build-id",
+ArgBuildID("build-id=",
            cl::desc("Request creation of \".note.gnu.build-id\" ELF note section."),
            cl::value_desc("style"));
 
-- 
1.8.1.3

